datafusion/physical-expr/src/equivalence/properties/mod.rs

```diff
@@ -255,10 +255,11 @@ impl EquivalenceProperties {
     pub fn constants(&self) -> Vec<ConstExpr> {
         self.eq_group
             .iter()
-            .filter_map(|c| {
-                c.constant.as_ref().and_then(|across| {
-                    c.canonical_expr()
-                        .map(|expr| ConstExpr::new(Arc::clone(expr), across.clone()))
+            .flat_map(|c| {
```

Comment:
The bug fix here is that this method now adds an entry for each equivalent expression rather than just the first "canonical" one

For example, if the equivalent expressions are `(a, b)` and the constant is known to be 1, now the code will return
`a=1, b=1` but before the fix it would only return `a=1`)